### PR TITLE
312 add GitHub vcs object

### PR DESCRIPTION
**Pull request by mbao01**
_Merge Order - 3_

### CODE REVIEW

**Comment by mbao01** [comment]
@yegor256 please review

File: objects/clients/github.rb [hunk 1]
```diff
@@ -49,5 +45,7 @@ def client
       puts "#{tp.defined_class}##{tp.method_id}()" if tp.defined_class == client.class
     end.enable
     client
+  rescue Octokit::NotFound
+    puts 'Issue with account not found'
```
Comment (yegor256) [inline objects/clients/github.rb:49]: @mbao01 this doesn't look like a nice error handling mechanism :)

File: objects/vcs/base.rb [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+#
+# AbstractVCS - provides a uniform interface to work with different VCSes
+#
+class AbstractVCS
```
Comment (yegor256) [inline objects/vcs/base.rb:24]: @mbao01 in Ruby we don't do this. This is what we do in Java, which is strongly typed.

**Comment by yegor256** [comment]
@mbao01 again, seems to be too many changes in one PR. You rename files/objects AND introduce new mechanism of error handling, which I don't really like. We can't catch exceptions and then just do `puts`. Google "exception swallowing" and why it's anti-pattern.

**Comment by mbao01** [comment]
@yegor256 my bad, I shouldn't have requested your review on this yet because it contains changes from #342  and #343.

There is a good reason for "swallowing" the exception here - users have encountered issues  where 0pdd fails because it cannot close an issue that does not exist. Yes, I agree with you. I should leave a puzzle there to be solved later instead of "swallowing" the exception. We need to catch the error and should have it logged or reported to an error tracking system.

File: objects/vcs/base.rb (hunk 2 above)
Comment (mbao01) [reply objects/vcs/base.rb:24]: Sure, will take it out. It is not required)

File: objects/vcs/github.rb [hunk 3]
```diff
@@ -0,0 +1,151 @@
+# Copyright (c) 2016-2021 Yegor Bugayenko
```
Comment (yegor256) [inline objects/vcs/github.rb:1]: @mbao01 this file is in this PR by mistake?

**Comment by yegor256** [comment]
@mbao01 see above

File: objects/vcs/github.rb (hunk 3 above)
Comment (mbao01) [reply objects/vcs/github.rb:1]: @yegor256 No, it is not. It will be used in the next PR. Trying to avoid a large PR.

**Comment by mbao01** [comment]
@yegor256 please review

File: objects/vcs/github.rb (hunk 3 above)
Comment (yegor256) [reply objects/vcs/github.rb:1]: @mbao01 we don't do this. we make sure that ALL changes required in a PR stay in the PR. and we don't put anything that doesn't belong to it.

Comment (mbao01) [reply objects/vcs/github.rb:1]: @yegor256 okay, I will close this one. I combined the changes here with the next PR.

**Comment by mbao01** [comment]
PR closed because the changes here is combined with #345

**Pull request closed without merging**